build.py

```diff
@@ -154,6 +205,27 @@ def get_package_type():
     log.debug('package type detected: {0}'.format(pt))
     return pt
 
+def get_package_dependencies(v):
+    deps = []
+
+    # dependencies on other externals packages    
+    if 'interdependencies' in v:
+        for d in v['interdependencies']:
+            deps.append(get_package_name(d))
+
+    # distro-specific dependencies
+    if 'distro_dependencies' in v:
+        distro_type = get_distribution_type()
+        distro_version = get_distribution_version()
+        v_distro_deps = v['distro_dependencies']
+        if distro_type in v_distro_deps:
+            distro_ver = get_distribution_version()
+            distro_type_deps = v_distro_deps[distro_type]
+            if distro_ver in distro_type_deps:
+                deps.extend(distro_type_deps[distro_ver])
```

Comment:
What happens if a distro dep is incorrect (e.g. contains a typo)?
Perhaps we should log/print a message when the conditionals on L221 and L224 aren't satisfied?

Reply:
I think that would just create noise for packages that don't have any dependencies for a given platform.

Reply:
Good point.

> What happens if a distro dep is incorrect (e.g. contains a typo)?

How about this question? I guess it could potentially block installation of the package?

Reply:
Actually, it might not be that much noise. I'll implement the logging

Reply:
> > What happens if a distro dep is incorrect (e.g. contains a typo)?
> 
> How about this question? I guess it could potentially block installation of the package?

Yeah it could. There's no validation of dependencies beyond what `fpm` already provides.

Reply:
Got it. We'd catch that ahead of time anyway.